crates/data-generator/src/blocks/output.rs

```diff
@@ -0,0 +1,628 @@
+                })
+                .collect();
+
+            Self {
+                prop_name: prop_name_pc,
+                prop_vars: prop_variants,
+                ordered,
+            }
+        }
+    }
+
+    impl ToTokens for PropertyEnum {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            let derive = if self.ordered {
+                quote!(Ord, PartialOrd,)
+            } else {
+                TokenStream::new()
+            };
+
+            let prop_name = self.prop_name.to_ident();
+
+            // "pub enum BubbleCoralFan"
+            tokens.extend(quote! {
+                #[derive(Debug, #derive Eq, PartialEq, Copy, Clone, Hash)]
+                pub enum #prop_name
+            });
+
+            let mut stream = TokenStream::new();
+            self.prop_vars
+                .iter()
+                .for_each(|var| var.to_tokens(&mut stream));
+
+            // "{ North, East, South, West, }"
+            tokens.append(Group::new(Delimiter::Brace, stream));
+        }
+    }
+
+    pub struct PropertyEnumVariant {
+        pub var_name: String,           // XIV
+        pub var_ordinal: Option<usize>, // Some(14)
+    }
+
+    impl ToTokens for PropertyEnumVariant {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            let struct_name = self.var_name.to_ident();
+            // "CampfireLit"
+            tokens.append(struct_name);
+
+            // "Moisture = 5"
+            self.var_ordinal.into_iter().for_each(|ord| {
+                tokens.append(Punct::new('=', Spacing::Alone));
+                tokens.append(Literal::usize_unsuffixed(ord));
+            });
+
+            // ","
+            tokens.append(Punct::new(',', Spacing::Alone));
+        }
+    }
+}
+
+mod prop_default {
+    use proc_macro2::{Ident, TokenStream};
+    use quote::{quote, ToTokens};
+
+    use super::StringExt as _;
+    use crate::blocks::model::StateProperty;
+
+    #[derive(Debug)]
+    pub struct PropertyDefault {
+        pub enum_name: Ident, // WeepingVines
+        pub def_var_name: Ident,
+    }
+
+    impl From<&StateProperty> for PropertyDefault {
+        fn from(def_prop: &StateProperty) -> Self {
+            PropertyDefault {
+                enum_name: def_prop.prop_enum.to_ident(),
+                def_var_name: def_prop.variant.to_ident(),
+            }
+        }
+    }
+
+    impl ToTokens for PropertyDefault {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            let enum_name = &self.enum_name;
+            let def_var_name = &self.def_var_name;
+
+            tokens.extend(quote! {
+                impl Default for #enum_name {
+                    fn default() -> Self {
+                        Self::#def_var_name
+                    }
+                }
+            });
+        }
+    }
+}
+
+mod prop_convert {
+    use proc_macro2::{Ident, Literal, TokenStream};
+    use quote::{quote, ToTokens};
+
+    use super::StringExt as _;
```

Comment:
Do we actually need to hide the ext-trait?

Reply:
Obsolete as of 4115128d4ceb2f4c64f815c965bd5728e6fff786.